Libraries/LibWeb/ContentSecurityPolicy/Violation.cpp

```diff
@@ -47,7 +47,7 @@ GC::Ref<Violation> Violation::create_a_violation_object_for_global_policy_and_di
 
     // 3. If global is a Window object, set violation’s referrer to global’s document's referrer.
     if (global_object) {
-        if (auto* window = dynamic_cast<HTML::Window*>(global_object.ptr())) {
+        if (auto* window = as_if<HTML::Window>(*global_object)) {
```

Comment:
You can remove the outer `if (global_object) {` if you keep using `global_object.ptr()` here